subprojects/scheduler/src/main/java/org/gradle/internal/scheduler/EdgeType.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.scheduler;
+
+public enum EdgeType {
+    /**
+     * Represents a dependency between two nodes, pointing from dependency node to dependent node.
+     * The edge is removed when the source node is completed.
+     */
+    DEPENDENCY_OF,
+
+    /**
+     * The target node can only be started once the source node has fully finished running.
+     * The edge is removed when the source node is completed.
+     */
+    MUST_COMPLETE_BEFORE,
+
+    /**
+     * The scheduler should try to avoid starting the target node before the source node is completed.
```

Comment:
Shouldn't that be a `SHOULD_START_BEFORE` edge? Why did we go back to using `SHOULD_COMPLETE_BEFORE`?